java/examples/src/java/org/apache/orc/examples/CompressionWriter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.orc.examples;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.exec.vector.BytesColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.LongColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.VectorizedRowBatch;
+import org.apache.orc.CompressionKind;
+import org.apache.orc.OrcFile;
+import org.apache.orc.TypeDescription;
+import org.apache.orc.Writer;
+
+import java.io.IOException;
+
+public class CompressionWriter {
+  public static void main(Configuration conf, String[] args) throws IOException {
+    TypeDescription schema =
+            TypeDescription.fromString("struct<x:int,y:string>");
```

Comment:
Please use 4-space indentation, @deshanxiao .
https://github.com/apache/orc/blob/9cf9d26a20ac4d43911d7967ecaef8ef26b60594/java/checkstyle.xml#L64

cc @guiyanakuang , the current rule seems to unable to prevent this case because we don't have `forceStrictCondition=true`.

Reply:
Sorry I didn't notice that. I re-verified the case of `forceStrictCondition=true`.
orc-core added 1206 Checkstyle violation. Mainly dangling method parameter definitions in strict mode are also treated as violations.
Like this.
```java
  public static Reader createReader(Path path,
                                    ReaderOptions options) throws IOException {
    return new ReaderImpl(path, options);
  }
```

Reply:
Ya, I also tried that and found it's too intrusive at this time, @guiyanakuang .